src/util/system.cpp

```diff
@@ -399,49 +312,43 @@ bool ArgsManager::ParseParameters(int argc, const char* const argv[], std::strin
         if (key.length() > 1 && key[1] == '-')
             key.erase(0, 1);
 
+        key.erase(0, 1);
+        std::string section;
+        util::SettingsValue value = InterpretOption(section, key, val);
         const unsigned int flags = FlagsOfKnownArg(key);
         if (flags) {
-            if (!InterpretOption(key, val, flags, m_override_args, error)) {
+            if (!CheckValid(key, value, flags, error)) {
                 return false;
             }
+            // Weird behavior preserved for backwards compatibility: command
+            // line options with section prefixes are allowed but ignored. It
+            // would be better if these options triggered the Invalid parameter
+            // error below.
+            if (section.empty()) {
```

Comment:
Hmmm I can't get from where this behavior is extracted ?

Reply:
re: https://github.com/bitcoin/bitcoin/pull/15934#discussion_r337810134

> Hmmm I can't get from where this behavior is extracted ?

This behavior makes no sense and was never intended, so it's buried away in current code. It happens because of the following code in `ArgsManager::ParseParameters`:

https://github.com/bitcoin/bitcoin/blob/9dd6bbba613d7462afdb6276c4002bc183478528/src/util/system.cpp#L411-L420

A section `section.name` command line setting here will fail to trigger the "Invalid parameter" error because `FlagsOfKnownArg` strips out the `section.` prefix and finds that `name` is a known setting. But `InterpretOption` fails to strip out the `section.` prefix and stores the setting in the `m_override_args` map under `section.name` instead of `name`, where it is ignored and never has any effect.

The new code improves this by documenting the behavior and making it easy to change in the future. It also avoids bugs like this by splitting the `section.name` string exactly once in `InterpretOption` instead of many different places and times.